src/mechanics/custom_widgets.rs

```diff
@@ -0,0 +1,168 @@
+use crate::visuals::settings_menu::*;
+use bevy::prelude::*;
+
+pub const WHITE: Color = Color::rgb(1.0, 1.0, 1.0);
+pub const DBROWN: Color = Color::rgb(0.49, 0.29, 0.14);
+//pub const LBROWN: Color = Color::rgb(0.72, 0.53, 0.36);
+
+#[derive(Component, Copy, Clone)]
+pub enum CountingSliderKeys {
+    Handle = 0,
+    Fill = 1,
+    Value = 2,
+    Decrement = 3,
+    Increment = 4,
+    Back = 5,
+}
+
+pub struct Slider {
+    pub back: (NodeBundle, CountingSliderKeys),
+    pub handle: (
+        ButtonBundle,
+        ButtonTypes,
+        SettingsMenuElements,
+        CountingSliderKeys,
+    ),
+    pub fill: (NodeBundle, CountingSliderKeys),
+}
+
+pub struct Spinner {
+    pub value_container: NodeBundle,
+    pub buttons_container: NodeBundle,
+    pub value: (TextBundle, SettingsMenuElements, CountingSliderKeys),
+    pub increment: (
+        ButtonBundle,
+        ButtonTypes,
+        SettingsMenuElements,
+        CountingSliderKeys,
+    ),
+    pub decrement: (
+        ButtonBundle,
+        ButtonTypes,
+        SettingsMenuElements,
+        CountingSliderKeys,
+    ),
+}
+
+pub fn create_widget_label(text: String) -> (TextBundle, SettingsMenuElements) {
+    (
+        TextBundle::from_section(
+            text,
+            TextStyle {
+                font_size: 20.0,
+                color: WHITE,
+                ..default()
+            },
+        ),
+        SettingsMenuElements::Text,
+    )
+}
+
+pub fn create_widget_slider() -> Slider {
+    Slider {
+        back: (
+            NodeBundle {
+                style: Style {
+                    width: Val::Percent(100.0),
+                    height: Val::Percent(20.0),
+                    align_items: AlignItems::Center,
+                    ..default()
+                },
+                background_color: bevy::prelude::BackgroundColor(DBROWN),
+                ..default()
+            },
+            CountingSliderKeys::Back,
+        ),
+        handle: (
+            ButtonBundle {
+                style: Style {
+                    width: Val::Percent(5.00),
+                    //Handle does not clip outside widget container because it is
+                    //parented to fill and subsequently back which is 20% of the
+                    //widget containers height. [200% of 20% = 40%]
+                    height: Val::Percent(200.00),
+                    ..default()
+                },
+                ..default()
+            },
+            ButtonTypes::Slider,
+            SettingsMenuElements::Button,
+            CountingSliderKeys::Handle,
+        ),
+        fill: (
+            NodeBundle {
+                style: Style {
+                    width: Val::Percent(0.0),
+                    height: Val::Percent(100.0),
+                    ..default()
+                },
+                background_color: bevy::prelude::BackgroundColor(WHITE),
```

Comment:
Since you've imported all items from bevy's prelude above with the use bevy::prelude::*, I'd recommend just changing this to the type itself: BackgroundColor(WHITE),

Reply:
Fixed, no idea why that was like that.